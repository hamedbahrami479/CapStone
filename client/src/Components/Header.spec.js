import React from 'react';
import { render, screen } from '@testing-library/react';
import HGIB from './Header';
import HgibStylesOne from './Header';
import HgibStylesTwo from './Header';

describe('Test for Header Component HGIB', () => {
  it('snapshot header component', () => {
    const { asFragment } = render(<HGIB />);
    expect(asFragment(<HGIB />)).toMatchSnapshot();
  });

  it('snapshots p tag in Header', () => {
    const { asFragment } = render(<HgibStylesOne />);
    expect(asFragment(<HgibStylesOne />)).toMatchSnapshot();

  }); it('snapshot second p tag in header', () => {
    const { asFragment } = render(<HgibStylesTwo />);
    expect(asFragment(<HgibStylesTwo />)).toMatchSnapshot();
  });
 /* it('should render header component with the right text', () => {
    const { getByText } = render(<HgibStylesTwo />);
    expect(getByText("Give It Back").toBeInTheDocument();
  });

  it('should render header component with the right text', () => {
    const { getByText } = render(<HgibStylesTwo />);
    expect(getByText(<HgibStylesTwo />)).not.toBeNull();
  });

 /* it('alternative way using toBeInTheDocument jest-dom utility library', () => {
    const { getByText } = render(<HGIB />);
    expect(getbyText).toBeInTheDocument();
  });*/
});
