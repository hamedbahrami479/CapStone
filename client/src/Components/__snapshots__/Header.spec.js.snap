// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header Component HGIB snapshot header component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot header component 2`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot header component 3`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot p component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot p tag in Header 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot p tag in Header component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot second p tag header component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot second p tag in header 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;

exports[`Test for Header Component HGIB snapshot second p tag in header component 1`] = `
<DocumentFragment>
  <div
    class="sc-dkzDqf coOZjv"
  >
    <h1
      class="sc-bczRLJ eUuSTf"
    >
      Give It Back 
    </h1>
    <h1
      class="sc-gsnTZi bGFuGh"
    >
      It\`s worthit
    </h1>
    <hr />
  </div>
</DocumentFragment>
`;
